### PR TITLE
add ssml and continuous speech recognition

### PR DESCRIPTION
**Pull request by wangrui-msft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java [hunk 1]
```diff
@@ -210,6 +214,32 @@ Mono<Response<Void>> recognizeWithResponseInternal(CallMediaRecognizeOptions rec
                     .setRecognizeOptions(recognizeOptionsInternal)
                     .setOperationContext(recognizeOptions.getOperationContext());
 
+                return contentsInternal.recognizeWithResponseAsync(callConnectionId, recognizeRequest, context);
+            } else if (recognizeOptions instanceof CallMediaRecognizeSpeechOptions) {
+                CallMediaRecognizeSpeechOptions speechRecognizeOptions = (CallMediaRecognizeSpeechOptions) recognizeOptions;
+
+                SpeechOptionsInternal speechOptionsInternal = new SpeechOptionsInternal().setEndSilenceTimeoutInMs(speechRecognizeOptions.getEndSilenceTimeoutInMs());
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java:221]: is this the only attribute SpeechOptionsInternal has?

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java [hunk 2]
```diff
@@ -210,6 +214,32 @@ Mono<Response<Void>> recognizeWithResponseInternal(CallMediaRecognizeOptions rec
...
+
+                RecognizeOptionsInternal recognizeOptionsInternal = new RecognizeOptionsInternal()
+                    .setSpeechOptions(speechOptionsInternal)
+                    .setInterruptPrompt(speechRecognizeOptions.isInterruptPrompt())
+                    .setTargetParticipant(CommunicationIdentifierConverter.convert(speechRecognizeOptions.getTargetParticipant()));
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java:226]: I wonder if this indentation is wrong or just the way Git presents it is wrong.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java [hunk 3]
```diff
@@ -210,6 +214,32 @@ Mono<Response<Void>> recognizeWithResponseInternal(CallMediaRecognizeOptions rec
...
+
+                recognizeOptionsInternal.setInitialSilenceTimeoutInSeconds((int) speechRecognizeOptions.getInitialSilenceTimeout().getSeconds());
+
+                PlaySourceInternal playSourceInternal = null;
+                if (recognizeOptions.getPlayPrompt() != null) {
+                    PlaySource playSource = recognizeOptions.getPlayPrompt();
+                    playSourceInternal = translatePlaySourceToPlaySourceInternal(playSource);
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java:233]: I think that `convert` is a better word than `translate`. Even `getPlaySourceInternal` may work better.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java [hunk 4]
```diff
@@ -272,6 +302,8 @@ PlayRequest getPlayRequest(PlaySource playSource, List<CommunicationIdentifier>
             playSourceInternal = getPlaySourceInternalFromFileSource((FileSource) playSource);
         } else if (playSource instanceof TextSource) {
             playSourceInternal = getPlaySourceInternalFromTextSource((TextSource) playSource);
+        } else if (playSource instanceof SsmlSource) {
+            playSourceInternal = getPlaySourceInternalFromSsmlSource((SsmlSource) playSource);
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java:306]: I like this approach to get the `playSourceInternal` variable. We should replicate it for `RecognizeRequest` (line 236) to stop that method from growing.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java [hunk 5]
```diff
@@ -322,6 +354,15 @@ private PlaySourceInternal getPlaySourceInternalFromTextSource(TextSource playSo
         return playSourceInternal;
     }
 
+    private PlaySourceInternal getPlaySourceInternalFromSsmlSource(SsmlSource playSource) {
+        SsmlSourceInternal ssmlSourceInternal = new SsmlSourceInternal().setSsmlText(playSource.getSsmlText());
+        PlaySourceInternal playSourceInternal = new PlaySourceInternal()
+            .setSourceType(PlaySourceTypeInternal.TEXT)
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java:360]: this should be SSML.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognitionType.java [hunk 6]
```diff
@@ -13,6 +13,10 @@ public final class CallMediaRecognitionType extends ExpandableStringEnum<CallMed
     public static final CallMediaRecognitionType DTMF = fromString("dtmf");
     /** Static value choices for RecognitionTypeInternal. */
     public static final CallMediaRecognitionType CHOICES = fromString("choices");
+    /** Static value continuous speech for RecognitionType. */
+    public static final CallMediaRecognitionType SPEECH = fromString("speech");
+    /** Static value continuous speech or dtmf for RecognitionTypeInternal. */
+    public static final CallMediaRecognitionType SPEECHORDTMF = fromString("speechordtmf");
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognitionType.java:19]: > The names of variables declared class constants and of ANSI constants should be all uppercase with words separated by underscores ("_")

https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognizeSpeechOptions.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** The Recognize configurations specific for Continuous Speech Recognition. **/
+public class CallMediaRecognizeSpeechOptions extends CallMediaRecognizeOptions {
+    /*
+     * The length of end silence when user stops speaking and cogservice send
+     * response.
+     */
+    @JsonProperty(value = "endSilenceTimeoutInMs")
+    private Long endSilenceTimeoutInMs;
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognizeSpeechOptions.java:16]: according to azure board we want timeouts to be `Duration`. 
Please see [DtmfOptions](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognizeDtmfOptions.java) for reference.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/RecognizeInputType.java [hunk 8]
```diff
@@ -15,6 +15,12 @@ public final class RecognizeInputType extends ExpandableStringEnum<RecognizeInpu
     /** Static value choices for RecognizeInputTypeInternal. */
     public static final RecognizeInputType CHOICES = fromString("choices");
 
+    /** Static value choices for RecognizeInputTypeInternal. */
+    public static final RecognizeInputType SPEECH = fromString("speech");
+
+    /** Static value choices for RecognizeInputTypeInternal. */
+    public static final RecognizeInputType SPEECHORDTMF = fromString("speechordtmf");
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/RecognizeInputType.java:22]: SPEECH_OR_DTMF

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognizeSpeechOptions.java (hunk 7 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognizeSpeechOptions.java]: changed to duration

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognitionType.java (hunk 6 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognitionType.java]: changed to underscore

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java (hunk 5 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java]: correct, changed to SSML

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java (hunk 4 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java:218]: yes, refactored the recognizeRequest methods

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java (hunk 3 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java]: changed to convert

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java (hunk 2 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java]: the indentation is checked through java checkstyle, which should be fine

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java (hunk 1 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallMediaAsync.java]: yes, SpeechOptionsInternal  is autogenerated

**Review by cochi2 — APPROVED** [review]

**Pull request merged**